packages/ingest/src/lock.ts

```diff
@@ -31,30 +43,54 @@ async function checkOldLocks(bucket, locks = []) {
 * @param {string} providerName - The provider name
 * @returns {integer} - Number of current locks in the bucket
 **/
-async function countLock(bucket, providerName) {
+export async function countLock(
+  bucket: string,
+  providerName: string
+): Promise<number> {
   const s3Objects = await listS3ObjectsV2({
     Bucket: bucket,
     Prefix: `${lockPrefix}/${providerName}`
   });
-  return checkOldLocks(bucket, s3Objects);
+
+  if (s3Objects === undefined) return 0;
+
+  const locks = <Lock[]>s3Objects.filter(({ Key }) => Key !== undefined);
+
+  return checkOldLocks(bucket, locks);
 }
 
-function addLock(bucket, providerName, filename) {
-  return s3PutObject({
+async function addLock(
+  bucket: string,
+  providerName: string,
+  filename: string
+): Promise<void> {
+  await s3PutObject({
     Bucket: bucket,
     Key: `${lockPrefix}/${providerName}/${filename}`,
     Body: ''
   });
 }
 
-function removeLock(bucket, providerName, filename) {
-  return deleteS3Object(
+export async function removeLock(
+  bucket: string,
+  providerName: string,
+  filename: string
+): Promise<void> {
+  await deleteS3Object(
     bucket,
     `${lockPrefix}/${providerName}/${filename}`
   );
 }
 
-async function proceed(bucket, provider, filename, counter = 0) {
+export async function proceed(
+  bucket: string,
+  provider: {
+    id: string,
+    globalConnectionLimit?: number
+  },
+  filename: string,
+  counter = 0
```

Comment:
Is this mixing type definition and default assignment?

Reply:
Yes. TS will determine the type based on the default.

Reply:
I guess so, but it's allowed when you're not destructuring a params object?

Reply:
Yup.